src/benchmarks/real-world/Roslyn/StageBenchmarks.cs

```diff
@@ -18,11 +20,36 @@ public class StageBenchmarks
         private CommonPEModuleBuilder _moduleBeingBuilt;
         private EmitOptions _options;
         private MemoryStream _peStream;
+        private ParseOptions _parseOptions;
+        private SourceText[] _files;
+
+        [GlobalSetup(Target = nameof(Parsing))]
+        public void ParsingSetup()
+        {
+            var projectDir = Environment.GetEnvironmentVariable(Helpers.TestProjectEnvVarName);
+            var responseFile = Path.Combine(projectDir, "repro.rsp");
+            var cmdLineParser = new CSharpCommandLineParser();
+            var args = cmdLineParser.Parse(new[] { "@" + responseFile }, projectDir, sdkDirectory: null);
+            _parseOptions = args.ParseOptions;
+            _files = new SourceText[args.SourceFiles.Length];
+            Parallel.For(0, _files.Length, index =>
+            {
+                using (var fstream = new FileStream(args.SourceFiles[index].Path, FileMode.Open))
+                {
+                    _files[index] = SourceText.From(fstream);
+                }
+            });
+        }
 
         [Benchmark]
-        public object Parsing()
+        public SyntaxTree[] Parsing()
         {
-            return Helpers.CreateReproCompilation();
+            var trees = new SyntaxTree[_files.Length];
+            Parallel.For(0, _files.Length, index =>
+            {
+                trees[index] = SyntaxFactory.ParseSyntaxTree(_files[index], _parseOptions);
+            });
+            return trees;
         }
```

Comment:
`Parallel.For` uses a heuristic that decides how many tasks to use to parallelize given code. This makes the benchmarks very often unstable (if you run it `n` times the results will most probably vary).

What is the point of making this benchmark parallel?

Maybe it would be good to have two implementations? Sequential and parallel?

Reply:
That probably makes sense. The simple answer here is: that's what we do in the compiler, so that's what I'm interested in measuring.